### PR TITLE
audit-fix: settle locked penalty might cause a revert if the bond is …

### PR DESCRIPTION
**Pull request by skhomuti**
…zero, so we need to skip this step

It broke assumption that the only case of zero settled amount is expired lock

### CODE REVIEW

File: src/abstract/CSBondCore.sol [hunk 1]
```diff
@@ -237,17 +237,17 @@ abstract contract CSBondCore is ICSBondCore {
...
+        // If no bond already
+        if (burnedShares == 0) return;
+
         IBurner(LIDO_LOCATOR.burner()).requestBurnShares(
             address(this),
             burnedShares
         );
-        burned = _ethByShares(burnedShares);
+        uint256 burned = _ethByShares(burnedShares);
         emit BondBurned(nodeOperatorId, _ethByShares(toBurnShares), burned);
```
Comment (dgusakov) [inline src/abstract/CSBondCore.sol:251]:
```suggestion
        emit BondBurned(nodeOperatorId, _ethByShares(toBurnShares), _ethByShares(burnedShares));
```

File: src/abstract/CSBondCurve.sol [hunk 2]
```diff
@@ -222,9 +222,11 @@ abstract contract CSBondCurve is ICSBondCurve, Initializable {
     /// @dev Reset bond curve for the given Node Operator to default.
     ///      (for example, because of breaking the rules by Node Operator)
     function _resetBondCurve(uint256 nodeOperatorId) internal {
-        _getCSBondCurveStorage().operatorBondCurveId[
-            nodeOperatorId
-        ] = DEFAULT_BOND_CURVE_ID;
+        CSBondCurveStorage storage $ = _getCSBondCurveStorage();
+        uint256 curveId = $.operatorBondCurveId[nodeOperatorId];
+        if (curveId == DEFAULT_BOND_CURVE_ID) return;
```
Comment (dgusakov) [inline src/abstract/CSBondCurve.sol:227]:
```suggestion
        if ($.operatorBondCurveId[nodeOperatorId] == DEFAULT_BOND_CURVE_ID) return;
```

**Review by dgusakov — DISMISSED** [review]

File: test/CSModule.t.sol [hunk 3]
```diff
@@ -5567,6 +5567,39 @@ contract CsmSettleELRewardsStealingPenaltyAdvanced is CSMCommon {
...
+
+        // settle all current bond to make and edge case when there is no bond but a new lock is applied
+        csm.reportELRewardsStealingPenalty(
+            noId,
+            blockhash(block.number),
+            amount
+        );
+        csm.settleELRewardsStealingPenalty(idsToSettle);
+
+        accounting.getBond(noId);
```
Comment (madlabman) [inline test/CSModule.t.sol:5588]: Assert is missing?

File: test/CSModule.t.sol [hunk 4]
```diff
@@ -5567,6 +5567,39 @@ contract CsmSettleELRewardsStealingPenaltyAdvanced is CSMCommon {
...
+    function test_settleELRewardsStealingPenalty_NoBond()
+        public
+        assertInvariants
+    {
+        uint256 noId = createNodeOperator();
+        uint256[] memory idsToSettle = new uint256[](1);
+        idsToSettle[0] = noId;
+        uint256 amount = accounting.getBond(noId) + 1 ether;
+
+        // settle all current bond to make and edge case when there is no bond but a new lock is applied
```
Comment (madlabman) [inline test/CSModule.t.sol:5580]: Maybe `accounting.penalize` is better?

File: test/CSModule.t.sol (hunk 3 above)
Comment (skhomuti) [reply test/CSModule.t.sol:5588]: debug call, removed

File: test/CSModule.t.sol (hunk 4 above)
Comment (skhomuti) [reply test/CSModule.t.sol]: yep

**Review by madlabman — APPROVED** [review]

**Review by dgusakov — APPROVED** [review]
